Automattic/wp-calypso
# Emails: Add informative message and actions when user can't add email to the selected domain

#### Proposed Changes

As shown here 1200182182542585-as-1201225902975505/f we should be more informative to the user when the user is not able to add an email subscription to the domain, when the user is not the owner of the domain. 

#### Testing Instructions

1. Transfer a domain to your site, where your user is not the owner, and there are no email subscriptions attached to it
2. go to /email/{domain}/purchase/{site}
3. Assert that you see this screen:

![image](https://user-images.githubusercontent.com/5689927/192509110-75db890b-5404-457b-ba33-d960ceaea8f8.png)

Assert the following statements:
1. The "log in" redirects you to the login page
2. The reach-out link is enabled if the domain is registered with us and has privacy enabled.
3. Each of the select buttons are disabled

Assert when clicking on reach out link that you see the following screen:
![image](https://user-images.githubusercontent.com/5689927/191758121-cb484864-96b0-4f43-8777-1eb4ecc1651a.png)

Assert that the links are logging the three different actions:
![image](https://user-images.githubusercontent.com/5689927/192751835-8da0bd1b-d40d-4630-976e-884d5b739e42.png)


#### Pre-merge Checklist






Related to #66650
